### PR TITLE
Add additional constructor patches

### PR DESCRIPTION
**Pull request by marcoesters**
Add additional constructor patches to:

* implement feedback in an upstream PR https://github.com/conda/constructor/pull/665
* fix bugs that can cause system-critical directories to be deleted: https://github.com/ContinuumIO/anaconda-issues/issues/11726

### CODE REVIEW

**Review by markan — COMMENTED** [review]
So what happens if we somehow have a new installer trying to work with an install created by the old installer? Do we need special code to handle this? While replacing .nonadmin with .admin is simple, do we want to make the new code either set .nonadmin or .admin as appropriate? Otherwise an old installer installed as admin and a new installer installed as nonadmin are hard to distinguish.

**Review by markan — COMMENTED** [review]
One other thought: if deletion is not an atomic, infallible operation, we should maybe take care to ensure these markers get deleted last.

**Comment by marcoesters** [comment]
> So what happens if we somehow have a new installer trying to work with an install created by the old installer? Do we need special code to handle this? While replacing .nonadmin with .admin is simple, do we want to make the new code either set .nonadmin or .admin as appropriate? Otherwise an old installer installed as admin and a new installer installed as nonadmin are hard to distinguish.

The `.nonadmin` patch will not be implemented. There are too many open questions about what this could do to already existing installers.

**Review by SC426 — APPROVED** [review]

**Review by pseudoyim — APPROVED** [review]

File: recipe/patches/0004-check-conda-path-uninstaller.patch [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
++        ${WordFind} $0 " " "E+$1" $2
++        IfErrors endloop_critical
++        ReadEnvStr $3 $2
++        StrCmp $3 $INSTDIR invalid_dir
++        IntOp $1 $1 + 1
++        goto loop_critical
++    endloop_critical:
++
++    # Primitive check to see that $INSTDIR points to a conda directory
++    StrCpy $0 "_conda.exe conda-meta\history Scripts\activate.bat"
```
Comment (jaimergp) [inline recipe/patches/0004-check-conda-path-uninstaller.patch:68]: Hey folks, I just noticed that `Scripts\activate.bat` might not be present if the installer is not shipping the `conda` package (which is technically possible, even if rare). You might want to adjust the patch for those using `constructor` from `defaults` to build non-conda installers. The other two files are guaranteed to exist. This change will be reflected in `constructor 3.4.4` anyway, scheduled for release this week.

**Pull request merged**